Merge Feature/app 50

WhiteMaken opened the pull request.

johann-tam committed:
#23 Update bluetooth conneection

Implement console logs

johann-tam committed:
#23 Update bluetooth connection

Make the car go forward on bluetooth connection

johann-tam committed:
#23 Implement button control

Implememts functionality of the buttons so that the forward button makes the car drive forward and the stop button stopping the car.

johann-tam committed:
#23 Update .gitignore

mikaelhoije committed:
#44 Implement speed slider button

mikaelhoije committed:
#44 Make pointer on bar start in the middle

WhiteMaken committed:
#44 Implement speed reverse mode

Cleaned up the logic of the seekbar. Made the seekbar discrete to have a more clear view on how to change speed. Added logic to change the behavior of the seekbar depending if the car is in reverse mode or not.

WhiteMaken committed:
#44 Fix hardcoded strings in activity_main.xml

Added strings in strings.xml so the strings are not hardcoded in activity_main.xml anymore.

WhiteMaken committed:
#44 Make code more robust

Added more string variable in strings.xml to replace hardcoded ones, and added comments for clarity purposes.

johann-tam committed:
#23 Update button control

Commit:
#23 Update button functionality

WhiteMaken committed:
#44 Modify the the formatting of the string sent via bluetooth

Changed the code so that the string sent via bluetooth will be now b+speed when the car is going backwards.

Commit:
#23 Update changed chars to strings

WhiteMaken committed:
#45 Implement steering slider

Added logic to the enable seekBar method to handle the steering bar, as well as added a steering bar and a corresponding text to the view.

WhiteMaken committed:
Merge with feature/app-23

# Conflicts:
#	arduinoRemote/app/src/main/java/com/example/arduinoremote/MainActivity.java

WhiteMaken committed:
#45 Update listeners for sliders

Changed the way the sliders send the messages and updated the gitignore to ignore gradle building files.

WhiteMaken committed:
#45 Change the format of the messages sent to the car

Added \n at the end of each bluetooth message, added safety nets so the program doesn't crash without bluetooth connection, added comments for clairty and modified forward and stop button to match the new design.
References #26 #27

WhiteMaken committed:
#51 Implement connection to bluetooth button

Added a button to the UI that allows you to connect to bluetooth, the button disappears if a connection is established.

WhiteMaken committed:
#50 Improve the look of the UI

Made the text for the two sliders more descriptive, added colors to the connect and change mode button and moved the steering bar to the center of the screen.

WhiteMaken committed:
#50 Fix a bug with the display of the steering bar

johann-tam commented on arduinoRemote/app/src/main/java/com/example/arduinoremote/ConnectBT.java:
```diff
@@ -3,32 +3,47 @@
     protected Void doInBackground(Void... devices) { //while the progress dialog is shown, the connection is done in background
         try {
-            if (btSocket == null || !isBtConnected) {
+            Log.d("Testing","Background");
```
we can remove all those lines since it works now

oscarhjern commented on arduinoRemote/app/src/main/java/com/example/arduinoremote/ConnectBT.java:
```diff
@@ -3,32 +3,47 @@
+
+                        if(device.getName().equals(carName)) {
+                            this.device = device;
+                            UUID uuid = UUID.fromString("00001101-0000-1000-8000-00805f9b34fb"); //Standard SerialPortService ID
```
Should it be hard-coded?  --> "00001101-0000-1000-8000-00805f9b34fb"

WhiteMaken replied:
Yes

johann-tam commented on arduinoRemote/app/src/main/java/com/example/arduinoremote/ConnectBT.java:
```diff
@@ -3,32 +3,47 @@
         }
-        catch (IOException e) {
-            ConnectSuccess = false;//if the try failed, you can check the exception here
+        catch (IOException ignored) {
```
Should we do something useful with exceptions instead of ignoring them ?

WhiteMaken replied:
Yes, but not sure what to do with them right now.

johann-tam commented on arduinoRemote/app/src/main/java/com/example/arduinoremote/MainActivity.java:
```diff
@@ -1,55 +1,168 @@
 
-    static String instruction = "s";
-    Button forwardBut, stopBut;
+    static String instruction = "v0";
```
Adding v0 as a constant for stopping

oscarhjern commented on arduinoRemote/app/src/main/java/com/example/arduinoremote/MainActivity.java:
```diff
@@ -1,55 +1,168 @@
+        //configuring steeringBar
+        steeringBar.setProgress(0);
+        steeringBar.setMin(-18);
+        steeringBar.setMax(18);
```
These are all hard-coded, should they?

WhiteMaken replied:
We think so, but we might look into it in more detail later

johann-tam commented on arduinoRemote/app/src/main/java/com/example/arduinoremote/MainActivity.java:
```diff
@@ -1,55 +1,168 @@
+                                                           accelText.setText(speedText);
+
+                                                           btAccelValue = (progress * 20);
+                                                           instruction = "v" + btAccelValue + "\n";
```
We could add a Strings file for all that commands

oscarhjern reviewed (changes requested):
Pls look at the comments in the code :)

WhiteMaken committed:
#50 Make the code more robust

Added constants to replace magic numbers to reduce eventual tech debt we would have to deal with later on.
